src/components/ECharts/ProtocolDNDChart/SortableChart.tsx

```diff
@@ -0,0 +1,966 @@
+import * as React from 'react'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+import dynamic from 'next/dynamic'
+import styled from 'styled-components'
+import { useDefiManager, useDarkModeManager } from '~/contexts/LocalStorage'
+import type { IChartProps } from '../types'
+import { LazyChart } from '~/layout/ProtocolAndPool'
+import { Denomination, Filters, FiltersWrapper, Toggle } from './Misc'
+import { useFetchAndFormatChartData } from './useFetchAndFormatChartData'
+import { EmbedChart } from '~/components/Popover'
+import { NftVolumeData } from '~/api/types'
+import { BAR_CHARTS } from '../ProtocolChart/utils'
+
+const AreaChart = dynamic(() => import('.'), {
+	ssr: false
+}) as React.FC<IChartProps>
+
+interface IProps {
+	protocol: string
+	color: string
+	historicalChainTvls: {}
+	chartDenominations: Array<{ symbol: string; geckoId?: string | null }>
+	bobo?: boolean
+	hallmarks?: Array<[number, string]> | null
+	geckoId?: string | null
+	chartColors: { [type: string]: string }
+	metrics: { [metric: string]: boolean }
+	activeUsersId: number | string | null
+	usdInflowsData: Array<[string, number]> | null
+	governanceApis: Array<string> | null
+	isHourlyChart?: boolean
+	isCEX?: boolean
+	tokenSymbol?: string
+	protocolId: string
+	twitterHandle?: string
+	nftVolumeData: NftVolumeData
+	enabled?: Record<string, string>
+}
+
+const CHART_TYPES = [
```

Comment:
the code in a few of these chart files looks like it's been copy pasted from other files, would be good to avoid that as it makes the code much harder to maintain

instead try to see if you can reduce that by increasing code sharing